odoo/addons/base/models/ir_actions.py

```diff
@@ -647,7 +716,45 @@ def create(self, vals_list):
                 parent = self.browse(parent_id)
                 vals['model_id'] = parent.model_id.id
                 vals['group_ids'] = parent.group_ids.ids
-        return super().create(vals_list)
+        actions = super().create(vals_list)
+
+        # create history entries
+        history_vals = []
+        for action, vals in zip(actions, vals_list):
+            if (code := vals.get("code")):
+                history_vals.append({"action_id": action.id, "code": code})
+        if history_vals:
+            self.env['ir.actions.server.history'].create(history_vals)
+
+        return actions
+
+    def write(self, vals):
+        if (new_code := vals.get('code')):
+            History = self.env["ir.actions.server.history"]
+            old_code = self.code
+            last_history = History.search([('action_id', '=', self.id)], limit=1)
+            if last_history:
+                last_history.active = False
+            else:
+                History.create({
```

Comment:
Why? When you create a record, you already created the historical version of it.